Sources/_StringProcessing/Algorithms/Matching/Matches.swift

```diff
@@ -310,7 +325,11 @@ extension RegexMatchesCollection: Collection {
     
     guard
       let start = searchIndex(after: currentMatch),
-      let nextMatch = try? regex.firstMatch(in: input[start...])
+      start <= searchBounds.upperBound,
+      let nextMatch = try? regex._firstMatch(
+        input,
+        subjectBounds: subjectBounds,
+        searchBounds: start..<searchBounds.upperBound)
```

Comment:
Elsewhere you mentioned adjusting current position instead of search bounds. Is that or does that need to be consistent here? When do we adjust one or the other?

Reply:
The `searchBounds` can be adjusted here, since it's a brand new search. The place we don't want to adjust `searchBounds` is within `_firstMatch`, since that's just moving forward the potential match's starting point.